b2sdk/sync/sync.py

```diff
@@ -216,16 +134,149 @@ def sync_folders(
+        except InvalidArgument as e:
+            raise CommandError('--%s %s', (e.field_name, e.message))
+
+
+class Synchronizer(object):
+    DELETE_MODE = 301
+    KEEP_DAYS_MODE = 302
+
+    KEEP_OR_DELETE_MODES = [DELETE_MODE, KEEP_DAYS_MODE]
+
+    def __init__(
+            self,
+            no_progress,
+            max_workers,
+            policies_manager=DEFAULT_SCAN_MANAGER,
+            dry_run=False,
+            allow_empty_source=False,
+            newer_file_mode=None,
+            keep_days_or_delete=None,
+            compare_version_mode=AbstractFileSyncPolicy.COMPARE_VERSION_MODTIME,
+            compare_threshold=0,
+            keep_days=None,
+    ):
+        """
+        Initialize synchronizer class and validate arguments
+
+        :param no_progress: if True, do not show progress
+        :type no_progress: bool
+        :param max_workers: max number of workers
+        :type max_workers: int
+        :param policies_manager: policies manager object
+        :param dry_run:
+        :type dry_run: bool
+        :param allow_empty_source: if True, do not check whether source folder is empty
+        :type allow_empty_source: bool
+        :param newer_file_mode: one of 101 (Skip) or 102 (Replace), can be None
+        :type newer_file_mode: int
+        :param keep_days_or_delete: one of 301 (Delete policy), 302 (keep for days policy), can be None
+        :type keep_days_or_delete: int
+        :param compare_version_mode: one of 201 (Modification time), 201 (Size), can be None, default is 201
+        :type compare_version_mode: int
+        :param compare_threshold: should be greater than 0, default is 0
+        :type compare_threshold: int
+        :param keep_days: if keep_days_or_delete is 302, then this should be greater than 0, default is None
+        :type keep_days: int
+        """
+        self.newer_file_mode = newer_file_mode
+        self.keep_days_or_delete = keep_days_or_delete
+        self.keep_days = keep_days
+        self.compare_version_mode = compare_version_mode
+        self.compare_threshold = compare_threshold
+        self.dry_run = dry_run
+        self.allow_empty_source = allow_empty_source
+        self.policies_manager = policies_manager
+        self.max_workers = max_workers
+        self.no_progress = no_progress
+        self._validate()
+
+    def _validate(self):
+        if self.compare_threshold < 0:
+            raise InvalidArgument('compare_threshold', 'compare_threshold (must be a positive integer')
+
+        if self.newer_file_mode and self.newer_file_mode not in AbstractFileSyncPolicy.NEWER_FILE_MODES:
+            raise InvalidArgument(
+                'newer_file_mode',
+                'newer_file_mode must be one of :%s' % AbstractFileSyncPolicy.NEWER_FILE_MODES,
+            )
+
+        if self.keep_days_or_delete and self.keep_days_or_delete not in self.KEEP_OR_DELETE_MODES:
```

Comment:
I think if `keep_days = None`, then files should be deleted immediately, and if keep_days is a positive integer, then it says how many days needs to pass. There should only be one argument, interface is then simple we can remove 10 lines of code for validation.